drivers/serial/uart_ite_it8xxx2.c

```diff
@@ -0,0 +1,931 @@
+#define DMACR_RX_BUF_1_FULL_INT_FG  BIT(7)
+#define DMACR_RX_BUF_0_FULL_INT_FG  BIT(6)
+#define DMACR_TX_DMA_FINISH_FG      BIT(1)        /* R/W0 */
+#define DMACR_RX_DMA_FINISH_FG      BIT(0)        /* R/W0 */
+#define DEF_ENABLE_CTS_RTS          FALSE
+#define DEF_UART2_ENABLE_CTS_RTS    TRUE          /* FALSE */
+#define DEF_UART1_ENABLE_CTS_RTS       FALSE    /* TRUE */
+#define IIRC(dev)                   (DEV_DATA(dev)->iir_cache)
+#define GPH1VS          BIT(1)
+#define GPH2VS          BIT(0)
+
+/**
+ * Const Variable
+ *
+ * GCR1 : uart enable
+ * GCR6 : each pin enable, BIT0:SIN, BIT1:SOUT
+ * GCR8 : total pin number
+ */
+static const struct T_UartStruct mComPortTb[t_uart_max_idx] = {
+	/* UART1 */
+	#if DEF_UART1_ENABLE_CTS_RTS
+	{ ((volatile struct T_ITEUART *)REG_UART1_BASE), &GPCRB0,
+	  &GPCRB1, &GPCRF3, &GPCRD5, (0x10 << 0), (BIT(0) | BIT(1)), 0x03 },
+	#else
+	{ ((volatile struct T_ITEUART *)REG_UART1_BASE), &GPCRB0,
+	  &GPCRB1, &GPCRF3, &GPCRD5, (0x01 << 0), (BIT(1)), 0x03 },
+	#endif
+
+	/* UART2 */
+	{ ((volatile struct T_ITEUART *)REG_UART2_BASE), &GPCRH1,
+	  &GPCRH2, &GPCRE5, &GPCRI7, (0x10 << 2), (BIT(2) | BIT(3)), 0x0C },
+};
+
+static const struct T_BaudrateStruct mBaudrateTb[t_br_max_idx] = {
+/**                       1,500,000
+ *              NUM      (3,000,000)         1
+ * ( (INT+1) + ----- ) = ----------- x --------------
+ *              DEN       baud rate    ((DLM<<8)|DLL)
+ *
+ *    INT, DEN, NUM,  DLL,  DLM
+ */
+	{ 155, 100,  25,   0x18, 0x00 },        // 0x00 - 4800
+	{ 155, 100,  25,   0x10, 0x00 },        // 0x01 - 9600
+	{ 103,   6,   1,   0x0c, 0x00 },        // 0x02 - 14400
+	{  77,   8,   1,   0x06, 0x00 },        // 0x03 - 19200
+	{  38,  16,   1,   0x03, 0x00 },        // 0x04 - 38400
+	{  25,  24,   1,   0x02, 0x00 },        // 0x05 - 57600
+	{  12,  48,   1,   0x01, 0x00 },        // 0x06 - 115200
+	{   5,  96,  49,   0x02, 0x80 },        // 0x07 - 230400
+	{   2, 192,  49,   0x01, 0x80 },        // 0x08 - 460800
+};
+
+/**
+ *
+ * uart_putc
+ *
+ * @return
+ *
+ * @parameter
+ *
+ *
+ */
+static uint8_t uart_putc(uint8_t chan, uint8_t data)
+{
+	unsigned long u32temp = 0;
+
+	while ((mComPortTb[chan].mUartBase->ULSR & LSR_THRE) == 0) {
+		if (u32temp == 0x100) {
+			return STATUS_FAIL;
+		}
+		u32temp++;
+	}
+	mComPortTb[chan].mUartBase->UTHR = data;
+	return STATUS_SUCCESS;
+}
+
+/**
+ *
+ * uart_getc
+ *
+ * @return
+ *
+ * @parameter
+ *
+ *
+ */
+static uint8_t uart_getc(uint8_t chan, uint8_t *data)
+{
+	unsigned long u32temp = 0;
+
+	while ((mComPortTb[chan].mUartBase->ULSR & LSR_DR) == 0) {
+		if (u32temp == 0x100) {
+			return STATUS_FAIL;
+		}
+		u32temp++;
+	}
+
+	*data = mComPortTb[chan].mUartBase->UTHR;
+	return STATUS_SUCCESS;
+}
+
+/**
+ *
+ * uart_is_xmit_empty
+ *
+ * @return
+ *
+ * @parameter
+ *
+ *
+ */
+static uint8_t uart_is_xmit_empty(uint8_t chan)
+{
+	return IS_MASK_SET(mComPortTb[chan].mUartBase->ULSR, LSR_TEMT);
+}
+
+/**
+ *
+ * uart_init
+ *
+ * @return
```

Comment:
Drop these comments if they done tell you anything useful. Better to fix them up